llvm/lib/Target/AMDGPU/AMDGPURBLegalize.cpp

```diff
@@ -0,0 +1,74 @@
+//===-- AMDGPURBLegalize.cpp ----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+/// Lower G_ instructions that can't be inst-selected with register bank
+/// assignment given by RB-select based on machine uniformity info.
```

Comment:
Nit: Maybe mention explicitly that this is expected to run after RB-select? This might be surprising since the usual GISel pipeline has legalization first and then RB-select.